src/shape/bounding-sphere.js

```diff
@@ -3,6 +3,7 @@ pc.extend(pc, function () {
     var tmpVecB = new pc.Vec3();
     var tmpVecC = new pc.Vec3();
     var tmpVecD = new pc.Vec3();
+    var diffBetweenPoints = new pc.Vec3();
```

Comment:
On a class these size, you're right, I should be using the temp variables here and will change them.

I'm just a bit conscious of a potential maintenance headache if/when the class gets bigger in regards to reusing global variables. e.g.

```
functionA : function() {
    tmpVecA.set(0, 0, 0);
}
```
Then on a later date, further down the file:
```
functionSomeOtherThing : function() {
    tmpVecA.set(1,2,3);
    // some more code
    functionA();
    var length = tmpVecA.length();
    // some more code
}
```
The code won't work correctly and it's a bit of a headache to debug due to the use of same global temp variables whereas having temp variables dedicated to a function is arguably more maintainable.

What do you think?